sdk-extensions/async-processor/src/main/java/io/opentelemetry/sdk/extension/trace/export/DisruptorEventQueue.java

```diff
@@ -205,11 +192,12 @@ public void onEvent(final DisruptorEvent event, long sequence, boolean endOfBatc
           case ON_START:
             @SuppressWarnings("unchecked")
             final SimpleImmutableEntry<ReadWriteSpan, Context> eventArgs =
-                (SimpleImmutableEntry<ReadWriteSpan, Context>) readableSpan;
+                (SimpleImmutableEntry<ReadWriteSpan, Context>)
+                    requireNonNull(readableSpan, "readableSpan");
```

Comment:
feels a bit scary to end up throwing exceptions here. What happens if this method throws?

Reply:
Went with null check pattern. Noticed that BSP will throw either way calling `span.getSpanContext`, though yeah probably something to fix there